src/library-authoring/components/ComponentCard.tsx

```diff
@@ -89,6 +94,76 @@ export const ComponentMenu = ({ usageKey }: { usageKey: string }) => {
   );
 };
 
+interface AddComponentWidgetProps {
+  usageKey: string;
+  blockType: string;
+}
+
+const AddComponentWidget = ({ usageKey, blockType }: AddComponentWidgetProps) => {
+  const intl = useIntl();
+
+  const {
+    componentPickerMode,
+    onComponentSelected,
+    addComponentToSelectedComponents,
+    removeComponentFromSelectedComponents,
+    selectedComponents,
+  } = useLibraryContext();
+
+  // istanbul ignore if: this should never happen
+  if (!usageKey) {
+    throw new Error('usageKey is required');
+  }
+
+  // istanbul ignore if: this should never happen
+  if (!componentPickerMode) {
+    return null;
+  }
+
+  if (componentPickerMode === 'single') {
+    return (
+      <Button
+        variant="outline-primary"
+        iconBefore={AddCircleOutline}
+        onClick={() => {
+          onComponentSelected({ usageKey, blockType });
+        }}
+      >
+        <FormattedMessage {...messages.componentPickerSingleSelectTitle} />
+      </Button>
+    );
+  }
```

Comment:
This code seems to be duplicated with the code in `ComponentInfo.tsx` ?

Reply:
This is really close. It's the button in the sidebar to mark the component as selected.

Reply:
Oh, I didn't know we had a button for that in the sidebar.

Reply:
I put it there for consistency. We can remove it later if the UX team doesn't feel it is necessary.